File: app/src/test/java/io/intrepid/skotlinton/screens/example1/Example1PresenterTest.java
package io.intrepid.skotlinton.screens.example1;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;

import io.intrepid.skotlinton.testutils.BasePresenterTest;
import io.intrepid.skotlinton.testutils.TestPresenterConfiguration;

import static org.mockito.Mockito.verify;

public class Example1PresenterTest extends BasePresenterTest<Example1Presenter> {

    @Mock
    Example1Contract.View mockView;

    @Before
    public void setup() {
        presenter = new Example1Presenter(mockView, TestPresenterConfiguration.createTestConfiguration());
    }

    @Test
    public void onButtonClick() throws Exception {
        presenter.onButtonClick();
        verify(mockView).gotoExample2();
    }
}

File: app/src/test/java/io/intrepid/skotlinton/screens/example1/Example1PresenterTest.kt
package io.intrepid.skotlinton.screens.example1

import org.junit.Before
import org.junit.Test
import org.mockito.Mock

import io.intrepid.skotlinton.testutils.BasePresenterTest
import io.intrepid.skotlinton.testutils.TestPresenterConfiguration

import org.mockito.Mockito.verify

class Example1PresenterTest : BasePresenterTest<Example1Presenter>() {

    @Mock
    internal var mockView: Example1Contract.View? = null

    @Before
    fun setup() {
        presenter = Example1Presenter(mockView!!, TestPresenterConfiguration.createTestConfiguration())
    }

    @Test
    @Throws(Exception::class)
    fun onButtonClick() {
        presenter!!.onButtonClick()
        verify<View>(mockView).gotoExample2()
    }
}
